cli/emit.rs

```diff
@@ -322,27 +333,55 @@ pub(crate) struct CheckEmitResult {
   pub stats: Stats,
 }
 
-/// Given a module graph, type check the module graph and optionally emit
-/// modules, updating the cache as appropriate. Emitting is determined by the
-/// `ts_config` supplied in the options, and if emitting, the files are stored
-/// in the cache.
+/// Given a set of roots and graph data, type check the module graph and
+/// optionally emit modules, updating the cache as appropriate. Emitting is
+/// determined by the `ts_config` supplied in the options, and if emitting, the
+/// files are stored in the cache.
 ///
 /// It is expected that it is determined if a check and/or emit is validated
 /// before the function is called.
 pub(crate) fn check_and_maybe_emit(
-  graph: Arc<ModuleGraph>,
+  roots: &[ModuleSpecifier],
+  graph_data: Arc<RwLock<GraphData>>,
   cache: &mut dyn Cacher,
   options: CheckOptions,
 ) -> Result<CheckEmitResult, AnyError> {
   let check_js = options.ts_config.get_check_js();
-  let root_names = get_root_names(&graph, check_js);
+  let segment_graph_data = {
+    let graph_data = graph_data.read();
+    graph_data.graph_segment(roots).unwrap()
+  };
+  if valid_emit(
+    &segment_graph_data,
+    cache,
+    &options.ts_config,
+    options.reload,
+    &options.reload_exclusions,
+  ) {
+    return Ok(Default::default());
+  }
+  let tsc_roots = get_tsc_roots(roots, &segment_graph_data, check_js);
+  if options.log_checks {
+    for root in roots {
+      let root_str = root.to_string();
+      // `$deno` specifiers are internal specifiers, printing out that
+      // they are being checked is confusing to a user, since they don't
+      // actually exist, so we will simply indicate that a generated module
+      // is being checked instead of the cryptic internal module
```

Comment:
Is it actually worth printing anything for generated modules? I'm not sure if generic `Check generated module` is any more useful

Reply:
Agreed